test/fail_compilation/test18484.d

```diff
@@ -0,0 +1,26 @@
+/* REQUIRED_ARGS: -dip1000
+TEST_OUTPUT:
+---
+fail_compilation/test18484.d(19): Error: returning `x.bar()` escapes a reference to local variable `x`
+fail_compilation/test18484.d(24): Error: escaping reference to stack allocated value returned by `S(0)`
+---
+*/
+
+// https://issues.dlang.org/show_bug.cgi?id=18484
+
+struct S
+{
+    int* bar() return;
+    int i;
+}
+
+int* test1()
+{
+    auto x = S(); return x.bar();  // error
+}
+
+int* test2()
+{
+    return S().bar();  // error
+}
```

Comment:
This code is not marked as `@safe`, how come it errors out ?

Reply:
Because it's obviously wrong. Even unsafe code gets some checks!

Reply:
So far, DIP1000 only affected `@safe` code (or at least that was the state intention). It looks like this change also change behavior in non-`@safe` code, which is very odd to me.